### ISSUE TITLE
another bug which crashes julia 1.0 and latest nightly 1.1.0

### ISSUE THREAD

**Issue by jmichel7** [JuliaLang/julia#29208]
put the following in a file bug.jl (it is an extract from code for multivariate polynomials which is reduced to a point where it is a bit nonsensical)

```
function Base.merge(a::Vector{Pair{K,V}},
                    b::Vector{Pair{K,V}})::Vector{Pair{K,V}} where K where V
   vcat(a,b)
end

struct Monomial{T}
  d::Vector{Pair{Symbol,T}}
end

struct Polynomial{T,N}
 d::Vector{Pair{Monomial{N},T}}
end

function Base.promote(a::Polynomial{T},b::Polynomial{T1}) where T where T1
  T2=promote_type(T,T1)
  if T2!=T  a=Polynomial([m=>convert(T2,c) for (m,c) in a.d]) end
  if T2!=T1 b=Polynomial([m=>convert(T2,c) for (m,c) in b.d]) end
  a,b
end

function bug(a::Polynomial, b::Polynomial)
  (a,b)=promote(a,b)
  u=merge(a.d,b.d)
end

a=Polynomial([Monomial([:a=>1])=>1])
u=Polynomial([m=>1//2*c for (m,c) in a.d])
bug(a,u)
```

then `julia bug.jl` gives:

```
Unreachable reached at 0x7f11c8a836a1

signal (4): Instruction non permise
in expression starting at /home/jmichel/julia/Bug.jl:31
bug at /home/jmichel/julia/Bug.jl:26
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1829
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:324
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:428
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:363 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:686
jl_interpret_toplevel_thunk_callback at /buildworker/worker/package_linux64/build/src/interpreter.c:799
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f11d3b312cf)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:808
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:787
jl_parse_eval_all at /buildworker/worker/package_linux64/build/src/ast.c:838
jl_load at /buildworker/worker/package_linux64/build/src/toplevel.c:821
include at ./boot.jl:317 [inlined]
include_relative at ./loading.jl:1038
include at ./sysimg.jl:29
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
exec_options at ./client.jl:229
_start at ./client.jl:421
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2182
jl_apply at /buildworker/worker/package_linux64/build/ui/../src/julia.h:1536 [inlined]
true_main at /buildworker/worker/package_linux64/build/ui/repl.c:112
main at /buildworker/worker/package_linux64/build/ui/repl.c:233
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/jmichel/julia-1.0.0/bin/julia (unknown line)
Allocations: 1003172 (Pool: 1002962; Big: 210); GC: 1
Instruction non permise (core dumped)

```

**Comment by JeffBezanson** [comment]
Fixed by #29380

**Comment by xukai92** [comment]
Since this issue is solved, should the message 

> Known issues
> The ScikitLearn SVM models will not work under Julia 1.0.3 but do work under Julia 1.1 due to Issue #29208

in the `README.md` be moved?
